mesonbuild/mtest.py

```diff
@@ -697,10 +697,10 @@ def preexec_fn() -> None:
 
         extra_cmd = []  # type: T.List[str]
         if self.test.protocol is TestProtocol.GTEST:
-            gtestname = '{}.xml'.format(self.test.name)
+            gtestname = '{}'.format(self.test.name)
```

Comment:
Either leave the .xml here, or replace this with `gtestname = self.test.name`

Reply:
Fixed